pkg/addon/addon.go

```diff
@@ -999,20 +1000,63 @@ func (h *Installer) getAddonMeta() (map[string]SourceMeta, error) {
 // installDependency checks if addon's dependency and install it
 func (h *Installer) installDependency(addon *InstallPackage) error {
 	var dependencies []string
+	var addonClusters = getClusters(h.args)
 	for _, dep := range addon.Dependencies {
-		_, err := FetchAddonRelatedApp(h.ctx, h.cli, dep.Name)
-		if err == nil {
-			continue
+		depApp, err := FetchAddonRelatedApp(h.ctx, h.cli, dep.Name)
+		var needInstallAddonDep = false
+		var depClusters []string
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				return err
+			}
+			// depApp is not exist
+			needInstallAddonDep = true
+			depClusters = addonClusters
+		} else {
+			// get the runtime clusters of current dependency addon
+			for _, r := range depApp.Status.AppliedResources {
+				if r.Cluster != "" && !stringslices.Contains(depClusters, r.Cluster) {
+					depClusters = append(depClusters, r.Cluster)
+				}
+			}
+
+			// determine if there are no dependencies on the cluster to be installed
+			for _, addonCluster := range addonClusters {
+				if !stringslices.Contains(depClusters, addonCluster) {
+					depClusters = append(depClusters, addonCluster)
+					needInstallAddonDep = true
+				}
+			}
 		}
-		if !apierrors.IsNotFound(err) {
-			return err
+		if !needInstallAddonDep {
+			continue
 		}
+
 		dependencies = append(dependencies, dep.Name)
 		if h.dryRun {
 			continue
 		}
 		depHandler := *h
-		depHandler.args = nil
+		// get dependency addon original parameters
+		var depArgs map[string]interface{}
+		var sec v1.Secret
+		err = h.cli.Get(h.ctx, client.ObjectKey{Namespace: types.DefaultKubeVelaNS, Name: addonutil.Addon2SecName(dep.Name)}, &sec)
+		if err == nil {
+			args, err := FetchArgsFromSecret(&sec)
+			if err == nil {
+				depArgs = args
+			}
+		}
+		// reset the cluster arg
+		if depArgs == nil {
+			depArgs = map[string]interface{}{
+				types.ClustersArg: depClusters,
```

Comment:
You can initialize this map beforehand, so you don't need to check if the map is nil here.